dirty_cat/minhash_encoder.py

```diff
@@ -172,56 +177,47 @@ def transform(self, X):
         """
         X = check_input(X)
         if self.minmax_hash:
-            assert self.n_components % 2 == 0, \
-                "n_components should be even when minmax_hash=True"
+            assert self.n_components % 2 == 0,\
+                    "n_components should be even when minmax_hash=True"
         if self.hashing == 'murmur':
-            assert not (self.minmax_hash), \
-                "minmax_hash not implemented with murmur"
+            assert not(self.minmax_hash),\
+                   "minmax_hash not implemented with murmur"
         if self.handle_missing not in ['error', 'zero_impute']:
             template = ("handle_missing should be either 'error' or "
                         "'zero_impute', got %s")
             raise ValueError(template % self.handle_missing)
 
-        # TODO Parallel run here
-        is_nan_idx = False
-
-        if self.hashing == 'fast':
-            X_out = np.zeros((len(X[:]), self.n_components * X.shape[1]))
-            counter = self.n_components
-            for k in range(X.shape[1]):
-                X_in = X[:, k].reshape(-1)
-                for i, x in enumerate(X_in):
-                    if isinstance(x, float):  # true if x is a missing value
-                        is_nan_idx = True
-                    elif x not in self.hash_dict:
-                        X_out[i, k * self.n_components:counter] = self.hash_dict[x] = self.get_fast_hash(x)
-                    else:
-                        X_out[i, k * self.n_components:counter] = self.hash_dict[x]
-                counter += self.n_components
-        elif self.hashing == 'murmur':
-            X_out = np.zeros((len(X[:]), self.n_components * X.shape[1]))
-            counter = self.n_components
-            for k in range(X.shape[1]):
-                X_in = X[:, k].reshape(-1)
-                for i, x in enumerate(X_in):
-                    if isinstance(x, float):
-                        is_nan_idx = True
-                    elif x not in self.hash_dict:
-                        X_out[i, k * self.n_components:counter] = self.hash_dict[x] = self.minhash(
+        if not (X == X).all():  # contains at least one missing value
+            if self.handle_missing == 'error':
+                raise ValueError("Found missing values in input data; set "
+                                 "handle_missing='zero_impute' to encode with missing values")
+            elif self.handle_missing == 'zero_impute':
+                X[~(X == X)] = "NAN"  # this will be replaced by a vector of zeroes by compute_hash
+
+        def compute_hash(x):
```

Comment:
Please put this function outside as a first level citizen and not nested inside. Making a function easy to important facilitates tests, reuse, and persistence.